src/Utilities/Utilities.jl

```diff
@@ -785,6 +785,45 @@ function check_strict_kw(strict)
     return nothing
 end
 
+"""
+Calls `git remote show \$(remotename)` to try to determine the main (development) branch
+of the remote repository. Returns `master` and prints a warning if it was unable to figure
+it out automatically.
+"""
+function git_remote_head_branch(varname, root; remotename = "origin", fallback = "master")
+    env = copy(ENV)
+    env["GIT_TERMINAL_PROMPT"] = "0"
+    env["GIT_SSH_COMMAND"] = get(env, "GIT_SSH_COMMAND", "ssh -o \"BatchMode yes\"")
+    cmd = `git remote show $(remotename)`
+    stderr_output = IOBuffer()
+    git_remote_output = try
+        read(pipeline(setenv(cmd, env; dir=root); stderr=stderr_output), String)
```

Comment:
@fredrikekre I removed the `ignorestatus` here, assuming that Git should return a non-zero status if `git remote show origin` is successful. Do you think this could break something?

Reply:
Probably fine since it is in `try-catch` since it will throw without `ignorestatus`, right?

Reply:
I believe so. I assume that if it exits with a non-zero, there will not be anything meaningful to parse from the stdout anyway, so better to bail here, than in the parsing step.